spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureConfigPropertySource.java

```diff
@@ -175,17 +183,16 @@ private String getKeyVaultEntry(String value) {
             // Check if we already have a client for this key vault, if not we will make
             // one
             if (!keyVaultClients.containsKey(uri.getHost())) {
-                KeyVaultClient client = new KeyVaultClient(uri);
+                KeyVaultClient client = new KeyVaultClient(uri, tokenCredentialProvider);
                 keyVaultClients.put(uri.getHost(), client);
             }
-            Duration keyVaultWaitTime = Duration.ofSeconds(appProperties.getKeyVaultWaitTime());
-            Secret secret = keyVaultClients.get(uri.getHost()).getSecret(uri, keyVaultWaitTime);
+            KeyVaultSecret secret = keyVaultClients.get(uri.getHost()).getSecret(uri, appProperties.getMaxRetryTime());
             if (secret == null) {
                 throw new IOException("No Key Vault Secret found for Reference.");
             }
             secretValue = secret.getValue();
         } catch (RuntimeException | IOException e) {
-            if (azureProperties.isFailFast()) {
+            if (!azureProperties.isFailFast()) {
```

Comment:
Wondering why the logic is flipped here? if fail fast, we are supposed to throw exception here?

Reply:
It was incorect in the previus release resulting in fail-fast doing the wrong thing.